contracts/DelegateRegistry.sol

```diff
@@ -11,22 +11,28 @@ contract DelegateRegistry {
     event SetDelegate(address indexed delegator, bytes32 indexed id, address indexed delegate, address previousDelegate);
     event ClearDelegate(address indexed delegator, bytes32 indexed id, address previousDelegate);
     
-    // delegate to an address
+    /// @dev Sets a delegate for the msg.sender and a specific id.
+    ///      The combination of msg.sender and the id can be seen as a unique key.
+    /// @param id Id for which the delegate should be set
+    /// @param delegate Address of the delegate
     function setDelegate(bytes32 id, address delegate) public {
         require (delegate != msg.sender, "Can't delegate to self");
         require (delegate != address(0), "Can't delegate to 0x0");
         address currentDelegate = delegation[msg.sender][id];
-        require (delegate != currentDelegate, "already delegated to this address");
+        require (delegate != currentDelegate, "Already delegated to this address");
         
```

Comment:
We should add `require (id != bytes32(0), "invalid ID")` since you cannot clear a delegation to `bytes32(0)` (the transaction always fails.

Reply:
Why should this fail? If I remember we never enforce that the id is not 0 (will write a test for this)